src/main/java/org/terasology/module/health/events/OnDamagedEvent.java

```diff
@@ -11,20 +12,16 @@
  */
 @OwnerEvent
 public class OnDamagedEvent extends OnHealthChangedEvent {
-    private Prefab damageType;
-    private int fullAmount;
+    private final Prefab damageType;
 
-    public OnDamagedEvent() {
-    }
-
-    public OnDamagedEvent(int fullAmount, int change, Prefab damageType, EntityRef instigator) {
-        super(instigator, change);
-        this.fullAmount = fullAmount;
+    public OnDamagedEvent(int change, Prefab damageType, EntityRef instigator) {
+        super(-change, instigator);
+        Preconditions.checkArgument(change >= 0, "damage amount must be non-negative - use OnRestoredEvent instead");
```

Comment:
should this arg check be done before calling `super()`? :thinking:

Reply:
the call to `super()` needs to be the first statement in the constructor. I'd like to have the condition check first, but that's not possible.

Reply:
okay, wasn't aware of that.